src/core/server/plugins/plugins_service.ts

```diff
@@ -60,10 +66,16 @@ export class PluginsService implements CoreService<PluginsServiceStart> {
 
     if (!config.initialize || this.coreContext.env.isDevClusterMaster) {
       this.log.info('Plugin initialization disabled.');
-      return new Map();
+      return {
+        contracts: new Map(),
+        uiPlugins: this.pluginsSystem.uiPlugins(),
```

Comment:
should we return list of uiPlugins if `initialize === false`? couldn't it be misinterpreted as they did initialized?

Reply:
Yep, this will be passed to the optimizer to build the bundles for the plugins. When running the optimizer in standalone (which we use to build distributables), initialization is disabled.